drivers/hmis_external_apis/lib/tasks/import.rake

```diff
@@ -115,4 +115,220 @@ namespace :import do
+      if !household
+        Rails.logger.info "Household not found for referral ID #{referral_id}. Skipping."
+        skipped_case_ids << case_id
+        next
+      end
+
+      household_id = Hmis::Hud::Base.generate_uuid
+      household.each do |member|
+        client = mci_id_to_client[member[:mci_id].to_s]&.source
+        if !client
+          skipped_mci_ids << member[:mci_id]
+          next # Skip but proceed with other members in the household
+        end
+
+        relationship_to_hoh = relation_to_id_mapping[member[:relation_to_id]] || 99
+        enrollment_id = Hmis::Hud::Base.generate_uuid
+        enrollment = Hmis::Hud::Enrollment.new(
+          ProjectID: project.ProjectID,
+          project_pk: project.id,
+          client: client,
+          EnrollmentID: enrollment_id,
+          EntryDate: entry_date_time.to_date,
+          RelationshipToHoH: relationship_to_hoh,
+          HouseholdID: household_id,
+          DateCreated: entry_date_time,
+          DateUpdated: entry_date_time,
+          UserID: system_hud_user.user_id,
+        )
+
+        case_id_to_hoh_enrollment[case_id] ||= enrollment
+        case_id_to_hoh_enrollment[case_id] = enrollment if enrollment.head_of_household?
+
+        enrollments << enrollment
+      end
+
+      # If couldn't find ANY household members for this household, log the CASE_ID as skipped
+      skipped_case_ids << case_id unless case_id_to_hoh_enrollment[case_id]
+    end
+
+    # Build CaseNotes
+    case_notes_export_file = HmisExternalApis::TcHmis::Importers::Loaders::XlsxFile.new(filename: "#{dir}/HCM-EXPORT-ContactNotes.xlsx", sheet_number: 0, header_row_number: 1)
+
+    case_notes = []
+    skipped_case_notes_ids = []
+    case_notes_export_file.each do |row|
+      case_id = row.field_value('CASEID')
+      information_date = row.field_value('CONTACT_DT')
+      note_content = row.field_value('CONTACT_NOTES')
+      contact_type = row.field_value('CONTACT_TYP_ID') # Append to note instead of storing in CDED, for simplicity
+      contact_type_str = contact_type_mapping[contact_type]
+      note_content << "\nContact Type: #{contact_type_str}" if contact_type_str
+      date_created = row.field_value('CREATED_DT')
+      date_updated = row.field_value('LST_UPDT_DT')
+      user_id = row.field_value('LST_UPDT_ID')
+
+      enrollment = case_id_to_hoh_enrollment[case_id]
+      if !enrollment
+        skipped_case_notes_ids << row.field_value('CONTACT_ID') # Enrollment not found, skip and record
+        next
+      end
+      case_notes << Hmis::Hud::CustomCaseNote.new(
+        CustomCaseNoteID: Hmis::Hud::Base.generate_uuid,
+        EnrollmentID: enrollment.EnrollmentID,
+        PersonalID: enrollment.PersonalID,
+        data_source_id: enrollment.data_source_id,
+        content: note_content,
+        UserID: user_id,
+        DateCreated: date_created,
+        DateUpdated: date_updated,
+        information_date: information_date.to_date,
```

Comment:
should these all be '.to_date' ?

Reply:
No, only Information Date is a date. The timsetamps are datetimes (which are in the file). Double-checked that these are being set correctly